app/models/job.rb

```diff
@@ -0,0 +1,16 @@
+class Job < ApplicationRecord
+  enum work_space_type: { onsite: 0, remote: 1, hybrid: 2 }
+
+  validates :title, :description, :company, :work_space_type, :location, :creator_id, :lead_source, presence: true
+  validates :title, uniqueness: true
+
+  belongs_to :department
+  belongs_to :creator, class_name: 'User', foreign_key: 'creator_id'
+
+  has_many :user_jobs, dependent: :destroy
+  has_many :users, through: :user_jobs
```

Comment:
Initially, I thought these are `creators` but there is a separate association name `creator`.
I think, `applicants` will be more suitable here :thinking: